File: src/main/java/alfheim/common/lexicon/page/PageTextLearnableKnowledge.java
package alfheim.common.lexicon.page;

import alexsocol.asjlib.ASJUtilities;
import alfheim.client.core.handler.CardinalSystemClient;
import alfheim.common.core.handler.CardinalSystem.KnowledgeSystem.Knowledge;
import net.minecraft.client.Minecraft;
import vazkii.botania.common.lexicon.page.PageText;

public class PageTextLearnableKnowledge extends PageText {

	final Knowledge knowledge;
	
	public PageTextLearnableKnowledge(String unName, Knowledge kn) {
		super(unName);
		knowledge = kn;
	}
	
	@SuppressWarnings("AccessStaticViaInstance")
	private boolean know() {
		if (Minecraft.getMinecraft().thePlayer == null) return false;
		return CardinalSystemClient.segment().knowledge[knowledge.ordinal()];
	}
	
	@Override
	public String getUnlocalizedName() {
		return ASJUtilities.isServer() || know() ? unlocalizedName : unlocalizedName + "u";
	}
}

File: src/main/java/alfheim/common/lexicon/page/PageTextLearnableKnowledge.kt
package alfheim.common.lexicon.page

import alexsocol.asjlib.*
import alfheim.client.core.handler.CardinalSystemClient.PlayerSegmentClient
import alfheim.common.core.handler.CardinalSystem.KnowledgeSystem.Knowledge
import vazkii.botania.common.lexicon.page.PageText

class PageTextLearnableKnowledge(unName: String, internal val knowledge: Knowledge): PageText(unName) {
	
	private fun know(): Boolean {
		return if (mc.thePlayer == null) false else PlayerSegmentClient.knowledge.contains(knowledge.toString())
	}
	
	override fun getUnlocalizedName(): String {
		return if (ASJUtilities.isServer || know()) unlocalizedName else unlocalizedName + "u"
	}
}
